authentication/src/main/java/io/camunda/authentication/filters/AdminUserCheckFilter.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.authentication.filters;
+
+import io.camunda.search.query.RoleQuery;
+import io.camunda.security.configuration.SecurityConfiguration;
+import io.camunda.service.RoleServices;
+import io.camunda.zeebe.protocol.record.value.DefaultRole;
+import io.camunda.zeebe.protocol.record.value.EntityType;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Map;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+public class AdminUserCheckFilter extends OncePerRequestFilter {
+
+  public static final String ADMIN_ROLE_ID = DefaultRole.ADMIN.getId();
+  public static final String USER_MEMBERS = "users";
+  private static final Logger LOG = LoggerFactory.getLogger(AdminUserCheckFilter.class);
+  private final SecurityConfiguration securityConfig;
+  private final RoleServices roleServices;
+
+  public AdminUserCheckFilter(
+      final SecurityConfiguration securityConfig, final RoleServices roleServices) {
+    this.securityConfig = securityConfig;
+    this.roleServices = roleServices;
+  }
+
+  @Override
+  protected void doFilterInternal(
+      final HttpServletRequest request,
+      final HttpServletResponse response,
+      final FilterChain filterChain)
+      throws ServletException, IOException {
+    if (request.getRequestURI().endsWith("/identity/setup")) {
+      filterChain.doFilter(request, response);
```

Comment:
is this kinda whitelisting the request when it goes to `/identity/setup`? does this mean when I already have an admin user, that I can still open the form?

Reply:
It means if someone navigates to `/identity/setup` it will ignore this specific filter and continue with the remainder of the security chain. In this case if someone is not logged in it will redirect them to the regular login page. Now that I think of it that might be a bit of an odd behavior. I'll probably take this out entirely.